src/cart/crt_swim.c

```diff
@@ -1053,12 +1074,15 @@ static int64_t crt_swim_progress_cb(crt_context_t crt_ctx, int64_t timeout_us, v
 				swim_net_glitch_update(csm->csm_ctx, self_id, delay);
 				csm->csm_last_unpack_hlc = hlc2;
 			}
+			crt_metrics_sample_delay(crt_ctx, delay, delay > max_delay);
```

Comment:
My understanding is that `delay` does not mean "network delay". It roughly indicates the duration from the most recent incoming request among all local `crt` contexts to the current time. Is that what you are after?

For something roughly related to "network delay", as seen by the `swim` context, subjected to clock offsets, may be `rcv_delay` in `crt_swim_srv_cb`.

Reply:
> Is that what you are after?

To be honest, I don't have a specific goal here other than to expose more of the inner workings of DAOS via telemetry. As a first step, I have been looking for anomalous log entries (e.g. the above "Network outage detected") that are often seen when things aren't going well, and trying to add telemetry that would allow an admin to correlate a spike on a graph with an increase in those log entries. The idea is to try to discern patterns in telemetry across a cluster rather than having to sift through logs.

To that end, if you or @frostedcmos have suggestions for more/better telemetry I could add here, please let me know.

Reply:
> [...] trying to add telemetry that would allow an admin to correlate a spike on a graph with an increase in those log entries.

Then I think the `delay` here is likely what you're after. Perhaps some name better than "network delay"? To throw a stone, say, "SWIM inactivity"?

Reply:
> To throw a stone, say, "SWIM inactivity"?

Hmm. At least for this first pass of adding telemetry, I am a little wary of trying to encode too much interpretation that results in the names of metrics diverging from the names in the code. That having been said, now that I step back and look at it a little, I agree that it would probably be best to specify that this metric is SWIM-specific, rather than a more generic measurement of networking performance. I'll update the name.